packages/aws-cdk-lib/aws-events-targets/test/appsync/appsync.test.ts

```diff
@@ -0,0 +1,394 @@
+import * as path from 'path';
+import { Template } from '../../../assertions';
+import * as appsync from '../../../aws-appsync';
+import * as events from '../../../aws-events';
+import * as iam from '../../../aws-iam';
+import * as sqs from '../../../aws-sqs';
+import * as cdk from '../../../core';
+import * as targets from '../../lib';
+
+const graphQLOperation = 'mutation Publish($message: String!){ publish(message: $message) { event } }';
+
+describe('AppSync GraphQL API target', () => {
+  let stack: cdk.Stack;
+  beforeEach(() => {
+    stack = new cdk.Stack();
+  });
+
+  test('fails when AWS_IAM auth is not configured', () => {
+    const noiam_api = new appsync.GraphqlApi(stack, 'noiamApi', {
+      name: 'no_iam_api',
+      definition: appsync.Definition.fromFile(path.join(__dirname, 'appsync.test.graphql')),
+    });
+
+    const rule = new events.Rule(stack, 'Rule', {
+      schedule: events.Schedule.rate(cdk.Duration.minutes(1)),
+    });
+
+    expect(() => {
+      rule.addTarget(new targets.AppSync(noiam_api, {
+        graphQLOperation,
+        variables: events.RuleTargetInput.fromObject({
+          message: events.EventField.fromPath('$.detail'),
+        }),
+      }));
+    }).toThrow('You must have AWS_IAM authorization mode enabled on your API to configure an AppSync target');
+  });
+
+  test('fails when graphQLEndpointArn is not configured', () => {
+    const noEndpointArnAPI = appsync.GraphqlApi.fromGraphqlApiAttributes(stack, 'ImportedAPI', {
+      graphqlApiId: 'MyApiId',
+      graphqlApiArn: 'MyApiArn',
+    });
+
+    const rule = new events.Rule(stack, 'Rule', {
+      schedule: events.Schedule.rate(cdk.Duration.minutes(1)),
+    });
+
+    expect(() => {
+      rule.addTarget(new targets.AppSync(noEndpointArnAPI, {
+        graphQLOperation,
+        variables: events.RuleTargetInput.fromObject({
+          message: events.EventField.fromPath('$.detail'),
+        }),
+      }));
+    }).toThrow('You must have AWS_IAM authorization mode enabled on your API to configure an AppSync target');
```

Comment:
is this the correct error to be checking for this test case? should be it expecting "throw new Error('You must have a valid `graphQLEndpointArn` set')" instead?

Reply:
right. fixing setup and check.